### DISCUSSION TITLE
2024: A Year That Matters – My Matter(.js) Review and Outlook

### DISCUSSION THREAD

**Discussion by Apollon77** [matter-js/matter.js#1591] [General]
# 2024: A Year That Matters – My Matter(.js) Review and Outlook

As 2024 comes to a close, it's exciting to reflect on an eventful year for the Matter standard and **matter.js**, the open-source JavaScript/TypeScript implementation of the Matter protocol. Since 2022, I’ve had the privilege of leading the []**matter.js** project](https://github.com/project-chip/matter.js) alongside a small team of open-source contributors. This year brought significant progress for Matter itself, our project, and the broader community. Here’s my personal perspective on 2024’s developments and what lies ahead.

---

## Evolution of the Matter Standard and Community  
This year saw the release of **Matter 1.3** and **Matter 1.4**, along with the started adoption by leading controller and device manufacturers. These updates introduced features like:  

- **Matter 1.3**: Energy measurement and new household device types.  
- **Matter 1.4**: Enhanced energy management, improved multi-admin mechanisms, and usability upgrades for Thread networks.

Beyond the headlines, critical "under-the-hood" enhancements have quietly made a difference. For instance, **Quieter Reporting Quality** in Matter 1.4 prevents overreporting in subscriptions, addressing real-world issues with devices like smart light bulbs, especially in Thread networks. These refinements demonstrate the commitment of the Connectivity Standards Alliance members to making Matter robust and future-proof.  

Additionally, **new re-certification options** aim to ease the process for manufacturers to update older devices to the latest standards. While this remains a significant effort, it’s a worthwhile step for end-users and the Matter ecosystem.  

One challenge that remains is addressing Thread-related interoperability issues. Controller manufacturers must collaborate for seamless end-user experiences. Transparency around controller capabilities and roadmaps would also help, though that might remain wishful thinking for now.  

Despite these hurdles, the vision for Matter remains strong. I believe it will play a pivotal role in the future of smart homes.

---

## **matter.js** Milestones in 2024  
This year marked several breakthroughs for **matter.js**, establishing it as a credible alternative implementation of the Matter standard.  

### Certified Success  
In a major milestone, **matter.js** powered its first **officially certified Matter device**. This proves that **matter.js** is not just a prototyping tool but a viable foundation for production-ready devices on OS-based platforms. Thanks to everyone who contributed to this achievement — let’s see what 2025 brings!  

### Connectivity Standards Alliance Member Meeting in Budapest  
November was another highlight, as I had the opportunity to showcase **matter.js** during a talk at the Connectivity Standards Alliance Member Meeting in Budapest. I demonstrated how **matter.js** simplifies compliant development for Matter devices and controllers, offering benefits for rapid prototyping and testing. Engaging with CSA members and working groups provided valuable insights to shape our roadmap.  

### Open-Source Adoption  
This year, several open-source projects embraced **matter.js** to create innovative solutions:  

- [**Matterbridge**](https://github.com/Luligu/matterbridge): A Homebridge-like plugin-based Matter bridge.  
- [**Home Assistant Matter Hub**](https://github.com/t0bst4r/home-assistant-matter-hub): A bridge for integrating Home Assistant entities with Matter.  
- [**Node-Red Matter Bridge**](https://github.com/sammachin/node-red-matter-bridge): Node-RED nodes for building Matter bridges.  
- **OpenHAB** and **ioBroker** are in Alpha/Beta stages for their Matter solutions, set for release in early 2025.  

These projects, collectively used by thousands, extend Matter’s reach by bridging legacy devices into the Matter ecosystem, paving the way for broader adoption.

### Key Releases in 2024  
We delivered four major **matter.js** updates this year:  

- **v0.8**: Introduced a high-level device-building API, enabling compliant devices with just seven lines of code.  
- **v0.9**: Added default implementations for key clusters like Level Control, Color Control and Window Coverings.  
- **v0.10**: Upgraded support to **Matter 1.3.0.1**, aligning with the latest cluster and protocol updates.  
- **v0.11**: Revamped the internal package structure, transitioned to the `@matter` NPM organization, and enhanced controller-building features.

Each release additionally included numerous fixes and optimizations, improving stability and compliance.

---

## Looking Ahead to 2025  
While 2024 was remarkable, the future holds even greater potential for matter.js and also Matter as a whole.  

- **Matter 1.4 support**: This is already in progress, alongside additional feature development.  
- **Expanded adoption**: With **OpenHAB** and **ioBroker** nearing release, we anticipate more feedback and contributions from the community.  
- **Certified devices**: We expect more devices to be built, certified, and released using **matter.js** as their foundation.  

On the protocol side, new Matter versions in 2025 will continue refining, optimizing, and expanding the standard. Together, we’ll bring Matter closer to its ultimate goal: a connected smart home in every household.

Here’s to an exciting 2025!

**Comment by Feuer-sturm** [comment]
A big thank you for your time and knowledge you invest in this great project. I am locking forward to the matter project 2025.
